Source/Core/VideoBackends/OGL/Render.cpp

```diff
@@ -1238,6 +1248,34 @@ void Renderer::ReinterpretPixelData(unsigned int convtype)
   }
 }
 
+void Renderer::SetFramebuffer(const AbstractFramebuffer* framebuffer)
+{
+  glBindFramebuffer(GL_FRAMEBUFFER, static_cast<const OGLFramebuffer*>(framebuffer)->GetFBO());
+  m_current_framebuffer = framebuffer;
+  m_current_framebuffer_width = framebuffer->GetWidth();
+  m_current_framebuffer_height = framebuffer->GetHeight();
+}
+
+void Renderer::SetAndDiscardFramebuffer(const AbstractFramebuffer* framebuffer)
+{
+  // EXT_discard_framebuffer could be used here to save bandwidth on tilers.
+  SetFramebuffer(framebuffer);
+}
+
+void Renderer::SetAndClearFramebuffer(const AbstractFramebuffer* framebuffer,
+                                      const ClearColor& color_value, float depth_value)
+{
+  SetFramebuffer(framebuffer);
+
+  // NOTE: This disturbs the current scissor setting.
+  // This won't be an issue when we implement proper state tracking.
+  glDisable(GL_SCISSOR_TEST);
+  if (framebuffer->GetColorFormat() != AbstractTextureFormat::Undefined)
+    glClearBufferfv(GL_COLOR, 0, color_value.data());
+  if (framebuffer->GetDepthFormat() != AbstractTextureFormat::Undefined)
+    glClearBufferfi(GL_DEPTH, 0, depth_value, 0);
```

Comment:
Just call glClear() with the correct bits. No need to clear all attachments after each other.

Reply:
I was using glClearBuffer() here to avoid changing the color/depth masks.

Reply:
Oh, nevermind, I was wrong, seems glClearBuffer() follows the same rules as glClear() IRT masking. I'll change it.